server/src-lib/Hasura/Server/Auth/JWT.hs

```diff
@@ -369,12 +368,12 @@ verifyJwt ctx (RawJWT rawJWT) = do
 
 
 instance J.ToJSON JWTConfig where
-  toJSON (JWTConfig ty keyOrUrl claimNs aud claimsFmt iss) =
+  toJSON (JWTConfig keyOrUrl claimNs aud claimsFmt iss) =
     case keyOrUrl of
          Left _    -> mkObj ("key" J..= J.String "<JWK REDACTED>")
          Right url -> mkObj ("jwk_url" J..= url)
     where
-      mkObj item = J.object [ "type" J..= ty
+      mkObj item = J.object [ "type" J..= J.String "<TYPE REDACTED>"
```

Comment:
This is bit clunky. It should be ideally with the `Left` pattern match right above. Hints on how to accomplish that?

Reply:
I’m afraid I don’t think I understand your question—could you elaborate?

Reply:
The `toJSON` would ideally not have `type` if `jwk_url` is provided. Currently, it was being serialized to have `type` for `jwk_url` as well. So now I need to send either two items ("key" and "type" on `Left`) or one item ("jwk_url" on Right).

Reply:
Perhaps something like this?

```haskell
instance J.ToJSON JWTConfig where
  toJSON (JWTConfig keyOrUrl claimNs aud claimsFmt iss) =
    case keyOrUrl of
      Left _    -> J.object $ [ "type" J..= J.String "<TYPE REDACTED>"
                              , "key" J..= J.String "<JWK REDACTED>"
                              ] ++ sharedFields
      Right url -> J.object $ ("jwk_url" J..= url) : sharedFields
    where
      sharedFields = [ "claims_namespace" J..= claimNs
                     , "claims_format" J..= claimsFmt
                     , "audience" J..= aud
                     , "issuer" J..= iss
                     ]
```

Reply:
…or perhaps this would be slightly nicer:

```haskell
instance J.ToJSON JWTConfig where
  toJSON (JWTConfig keyOrUrl claimNs aud claimsFmt iss) =
    J.object (jwkFields ++ sharedFields)
    where
      jwkFields = case keyOrUrl of
        Left _    -> [ "type" J..= J.String "<TYPE REDACTED>"
                     , "key" J..= J.String "<JWK REDACTED>" ]
        Right url -> [ "jwk_url" J..= url ]
      sharedFields = [ "claims_namespace" J..= claimNs
                     , "claims_format" J..= claimsFmt
                     , "audience" J..= aud
                     , "issuer" J..= iss
                     ]
```

Reply:
Haha. Thanks. I guess there ain't no magic to do this with one or two symbols xD